lib/browser/api/menu.js

```diff
@@ -164,10 +164,15 @@ Menu.buildFromTemplate = function (template) {
   const sorted = sortTemplate(filtered)
 
   sorted.forEach((item) => {
-    if (typeof item !== 'object') {
+    if (typeof item !== 'object' && item != null) {
       throw new TypeError('Invalid template for MenuItem')
     }
-    menu.append(new MenuItem(item))
+    if (item != null && Object.keys(item).length > 0) {
```

Comment:
I don't think it's right to check `Object.keys` here—if the user passes the empty object, that's an invalid template, and that case will get caught by the `!item.label && !item.role && !item.type` check you have below.

Reply:
Okay so in that case do you think we should invalidate (as in throwing errors) `null` items as well if we're invalidating empty objects.